src/PhpSpec/Console/Application.php

```diff
@@ -57,9 +58,17 @@ public function getContainer()
      */
     public function doRun(InputInterface $input, OutputInterface $output)
     {
+        $helperSet = $this->getHelperSet();
         $this->container->set('console.input', $input);
         $this->container->set('console.output', $output);
-        $this->container->set('console.helper.dialog', $this->getHelperSet()->get('dialog'));
+        $this->container->set('console.helper.dialog', $helperSet->get('dialog'));
```

Comment:
I was thinking of the case where someone extends or decorates the DialogHelper and tries to inject their modified version via the container. Maybe we don't care about that.